apps/client/src/app/core/scheme/criteria/criteria.component.ts

```diff
@@ -0,0 +1,211 @@
+  }
+  showDialogEdit(criteria: Criteria) {
+    this.visibleEdit = true
+    this.criteriaEdit = criteria
+    this.formGroup.controls.subIndex.setValue(this.criteriaEdit.subIndex)
+    this.formGroup.controls.name.setValue(this.criteriaEdit.name)
+    this.formGroup.controls.alias.setValue(this.criteriaEdit.alias)
+    this.formGroup.controls.requiresEvidence.setValue(
+      this.criteriaEdit.requiresEvidence
+    )
+    this.formGroup.controls.helpText.setValue(this.criteriaEdit.helpText)
+  }
+  confirmationDelete(event: Event, id: number, name: string) {
+    this.confirmationService.confirm({
+      target: event.target as EventTarget,
+      message: `¿Estás seguro de que quieres eliminar el criterio <strong>${name}</strong>?`,
+      header: 'Eliminar criterio',
+      icon: 'pi pi-info-circle',
+      acceptButtonStyleClass: 'p-button-danger p-button-text',
+      rejectButtonStyleClass: 'p-button-text p-button-text',
+      acceptIcon: 'none',
+      rejectIcon: 'none',
+      acceptLabel: 'Sí',
+      rejectLabel: 'No',
+
+      accept: () => {
+        this.toast.show('info', 'Eliminando..', 'Eliminando criterio..')
+        this.onDelete(id)
+      },
+      reject: () => {
+        this.toast.show('error', 'Rechazado', 'Haz rechazado la eliminación')
+      }
+    })
+  }
+  getCriterionPerIndicator(indicatorId: number) {
+    this.criteriaService
+      .getAllCriterionByIndicatorIndex(indicatorId)
+      .subscribe({
+        next: (res) => {
+          this.criterion = res.status === 'success' ? res.data : this.criterion
+        },
+        error: (e) => {
+          console.error(e)
+          this.toast.show('error', 'Error', e.error.data.message)
+        }
+      })
+  }
+  onDelete(id: number) {
+    this.criteriaService.delete(id).subscribe({
+      next: () => {
+        this.toast.show('success', 'Eliminado', 'Criterio eliminado con éxito')
+        this.getCriterionPerIndicator(this.indicatorIndex)
+      },
+      error: (e) => {
+        console.error(e)
+        this.toast.show('error', 'Error', e.error.data.message)
+      }
+    })
+  }
+  onCreate() {
+    this.fetchCreateCriterion = true
+    const { subIndex, name, alias, requiresEvidence, helpText } =
+      this.formGroup.controls
+    const criteria: CriteriaDTO = {
+      indicatorIndex: this.indicatorIndex,
+      subIndex: subIndex.value,
+      name: name.value,
+      alias: alias.value,
+      requiresEvidence: requiresEvidence.value,
+      helpText: helpText.value
+    }
+    this.criteriaService.create(criteria).subscribe({
+      next: () => {
+        this.fetchCreateCriterion = false
+        this.toast.show('success', 'Creado', 'Criterio creado con éxito')
+        this.closeDialog()
+        this.getCriterionPerIndicator(this.indicatorIndex)
+      },
+      error: (e) => {
+        this.fetchCreateCriterion = false
+        this.toast.show('error', 'Error', e.error.data.message)
+        this.closeDialog()
+      }
+    })
+  }
+  onEdit() {
```

Comment:
No estás chequeando que el form sea válido

```suggestion
  onEdit() {
    if (this.formGroup.invalid) return
    ...
```